templates/CRM/Event/Page/EventInfo.tpl

```diff
@@ -188,7 +188,7 @@
                           <td class="{$lClass} crm-event-label">{$feeBlock.label.$idx}</td>
                           {if $isPriceSet & $feeBlock.isDisplayAmount.$idx}
             <td class="fee_amount-value right">
-                              {if isset($feeBlock.tax_amount.$idx)}
+                              {if $feeBlock.tax_amount && $feeBlock.tax_amount.$idx}
```

Comment:
Can tax amount legitimately be 0 or would that make no sense? Or maybe it's not too important anyway since it just means it wouldn't display a 0, although was the intent to explicitly show 0?

Reply:
I think the intent would be to not show a 0 - I'm sure it could be argued either way

Reply:
also is it not the case here that $fee_block.tax_amount would be an array given we are looking at a specific key within an array?

Reply:
It seems ok with it being an array. In fact when compiled ALL smarty vars are array elements, i.e. `{$x}` compiles to `$this->_tpl_vars['x']`.

Reply:
For 0, it seems it displays it anyway because what it actually displays in both the `if` and the `else` is the `value` member. And that's actually a bigger string not just the number (see screenshot), although it then does some weird double currency symbol'ing because of the `|crmMoney` in the `else`. But I'm going to merge because it seems like an edge case that doesn't make sense - why would you pick a financial type for the price set item that has a tax account associated if you don't want to charge tax on that item.

<img width="336" alt="Untitled" src="https://user-images.githubusercontent.com/2967821/145586508-eabb0d90-b4aa-4664-8420-617f074028a6.png">